File: EntitySystem2/AttachSystem.cpp
#include "AttachSystem.h"

void AttachSystem::ProcessEntity(EntityW::EntitySp entity, EntityW::Time deltaTime) {
	auto transform = entity->get<TransformComponent>();
	auto attach = entity->get<AttachComponent>();

	transform->position = attach->parentTransform->position + attach->relativePosition;
}

void AttachSystem::OnComponentAttached(EntityW::EventSp event) {
	BaseSystem::OnComponentAttached(event);
}

File: EntitySystem2/scripts/attachSystem.lua
return {
	requiredComponents = { Components.Transform, Components.Attach },
	processEntity = function(self, entity, deltaTime)
		local transform = entity:get(Components.Transform)
		local attach = entity:get(Components.Attach)
		transform.position.x = attach.parentTransform.position.x + attach.relativePosition.x
		transform.position.y = attach.parentTransform.position.y + attach.relativePosition.y
	end
}